### DISCUSSION TITLE
Docking on Molstar

### DISCUSSION THREAD

**Discussion by projectssimm** [molstar/molstar#1271] [Q&A]
Hi guys, I am working on a molecular docking project that involves interaction with a 3D interface. The following needs should be met:

1. When a user uploads a complex file, I would like to automatically determine the center coordinates based on the ligand within the file.
2. Additionally, the user should also have the option to click on the 3D view to determine or modify the coordinates. In this case, the user should be able to click on an atom, molecule, or residue to retrieve the corresponding coordinates or the center coordinates.
3. Is there an API document for mol* available? Right now I cannot find any online.

Best,

**Comment by QDMarkMan** [comment]
For the first requirement, you can try focusing on the `Loci` of your custom ligand. Here’s an example from [pdb-molstar](https://github.com/molstar/pdbe-molstar/blob/570773fbf057d35262169308aa45797256154dbb/src/app/viewer.ts#L692) that demonstrates the focus method. It may contain some useful information for you.
